services/cd-service/pkg/repository/repository.go

```diff
@@ -382,14 +407,18 @@ func (r *repository) ProcessQueueOnce(e element) {
 			if err != nil || len(elements) == 0 {
 				return
 			}
-			if pushErr := r.Push(e.ctx, pushAction); pushErr != nil {
+			if pushErr := r.Push(e.ctx, pushAction(pushOptions, r)); pushErr != nil {
 				err = &InternalError{inner: pushErr}
 			}
 		} else {
-			err = &InternalError{inner: err}
+			logger.Error(fmt.Sprintf("error while pushing: %s", err))
+			err = httperrors.PublicError(ctx, errors.New(fmt.Sprintf("could not push to manifest repository '%s' on branch '%s' - this indicates that the ssh key does not have write access", r.config.URL, r.config.Branch)))
```

Comment:
Most callers are not http but grpc. How will they display this error?

Reply:
the package name is misleading. It is actually returning the error fine for grpc, I tested it with `evans`.